src/tasks.js

```diff
@@ -260,8 +263,11 @@ function rerunHttpRequestSend (options, transaction, callback) {
   logger.info(`Rerun Transaction #${transaction._id} - HTTP Request is being sent...`)
   const req = http.request(options, (res) => {
     res.on('data', chunk => {
-      // response data
-      response.body += chunk
+      /*
+       *  Don't need the response body at this point, because it's already been captured 
+       *  in GridFS (from router.js). 
+       *  Still need to have 'data' listener defined, or it changes program behaviour
+       */
```

Comment:
Im not sure im fully understanding this. Why arent we doing anything with this request data coming back? or rather, why is it being processed here and somewhere else so that we dont need the response here

Reply:
The rerun task is a wrapper for a routed transaction. The routed transaction is always stored in GridFS. But for reruns, there is an extra (higher) layer where the task runner submits a request to the rerun port, and gets a response back on the rerun port. What this area of the code is capturing is the response from the rerun port, which at this point has already been sent to gridFS as the routed transaction response. This response is discarded further up the call stack anyway (in the non-streaming version of the core); we're just not accumulating all the chunks here and taking the memory hit.

Reply:
ok great that makes sense. So this is the response from the rerun request (Request sent to rerun port).